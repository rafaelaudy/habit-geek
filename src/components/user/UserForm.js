import React, { Fragment } from "react";
import { useTranslation } from "react-i18next";

const UserForm = ({ email, password, setEmail, setPassword }) => {
  const { t } = useTranslation();

  return (
    <Fragment>
      <div className="mb-3">
        <label htmlFor="user-form-email">{t("user-form-email")}</label>
        <input
          id="user-form-email"
          className="form-control"
          value={email}
          onChange={e => setEmail(e.target.value)}
          required
        ></input>
        <div className="invalid-feedback">
          {t("user-form-email-validation")}
        </div>
      </div>
      <div className="mb-3">
        <label htmlFor="user-form-password">{t("user-form-password")}</label>
        <input
          id="user-form-password"
          className="form-control"
          value={password}
          onChange={e => setPassword(e.target.value)}
          required
        ></input>
        <div className="invalid-feedback">
          {t("user-form-password-validation")}
        </div>
      </div>
    </Fragment>
  );
};

export default UserForm;
