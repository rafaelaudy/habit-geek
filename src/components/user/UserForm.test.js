import React from "react";
import { shallow } from "enzyme";
import UserForm from "./UserForm";

//TODO
describe("UserForm component", () => {
  it("Renders static elements", () => {
    const userForm = shallow(<UserForm />);
    expect(userForm).toMatchSnapshot();
  });
});
