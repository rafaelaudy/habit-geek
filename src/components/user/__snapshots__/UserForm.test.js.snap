// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm component Renders static elements 1`] = `
<Fragment>
  <div
    className="mb-3"
  >
    <label
      htmlFor="user-form-email"
    >
      user-form-email
    </label>
    <input
      className="form-control"
      id="user-form-email"
      onChange={[Function]}
      required={true}
    />
    <div
      className="invalid-feedback"
    >
      user-form-email-validation
    </div>
  </div>
  <div
    className="mb-3"
  >
    <label
      htmlFor="user-form-password"
    >
      user-form-password
    </label>
    <input
      className="form-control"
      id="user-form-password"
      onChange={[Function]}
      required={true}
    />
    <div
      className="invalid-feedback"
    >
      user-form-password-validation
    </div>
  </div>
</Fragment>
`;
